server.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+	emailLib "github.com/jordan-wright/email"
+	_ "github.com/lib/pq"
+)
+
+const (
+	contentType     = "Content-Type"
+	jsonContentType = "application/json;charset=UTF-8"
+)
+
+func NewServer(httpAddr string, db *sql.DB, emailPool *emailLib.Pool) *http.Server {
+	// TODO - Add logging middleware
+	// TODO - Add secure headers middleware
+	r := mux.NewRouter()
+
+	r.HandleFunc("/api/v1/email", CreateEmailAccountHandler(db)).Methods("POST")
+	r.HandleFunc("/api/v1/email/{id}/send", SendEmailHandler(db, emailPool)).Methods("POST")
+	r.HandleFunc("/api/v1/email/bulksend", SendBulkEmailHandler(db, emailPool)).Methods("POST")
+	http.Handle("/", r)
+
+	return &http.Server{
+		Addr:    httpAddr,
+		Handler: r,
+	}
+}
+
+func readReqBody(r *http.Request) ([]byte, error) {
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+	if err != nil {
+		log.Errorf("Error occured when reading r.Body: %s", err)
+		return []byte{}, err
```

Comment:
@ajvb Any reason not to do `return nil, err` ?

Reply:
fixed